Separate schedules for mixed mode and normal mode in upgrade

SaitTalhaNisanci opened the pull request:
This is done so that we can have tests for only one of the modes instead of both of them.

marcocitus commented:
What are mixed and normal mode?

SaitTalhaNisanci commented:
in mixed mode one the workers is not upgraded.

marcocitus reviewed (approved):
These test scripts could really use some comments, e.g. I could not easily figure out what was meant by "mixed mode" or "local execution".

SaitTalhaNisanci commented:
@marcocitus You are right, local execution seems to be pretty confusing especially when we have 'local execution' concept in the engine.

I have changed it as `IsRunningOnLocalMachine` and added some comment. For the mixed mode, we have a comment that is explaining the argument:

```
--mixed                                 Run the verification phase with one node not upgraded
```
If you have any better wording etc I can update this place as well.
